salesforce/volunteer_job.py

```diff
@@ -1,46 +1,52 @@
-from .client import SalesforceClient
 import json
 import requests
 import threading
+from .client import SalesforceClient
+from common.helpers.queue import enqueue
+from common.models import Tag
 
 ''' ProjectPosition model maps to the Volunteer Job object in Salesforce '''
 client = SalesforceClient()
 
 
 def run(request):
-    response = SalesforceClient().send(request)
-
-
-def save(project_position: object):
-    """
-        POST to the composite endpoint
-    """
-    data = {
-        "allOrNone": "true",
-        "compositeRequest": [
-            {
-                "method": "GET",
-                "referenceId": "MyCampaign",
-                "url": f'{client.campaign_endpoint}/platform_id__c/{project_position.position_project.id}'
-            },
-            {
-                "method": "POST",
-                "referenceId": "ThisJob",
-                "url": client.job_endpoint,
-                "body": {
-                    "platform_id__c": project_position.id,
-                    "GW_Volunteers__Campaign__c": "@{MyCampaign.Id}",
-                    "name": "Front-End Developer",
-                    "gw_volunteers__description__c": "Volunteer position"
-                }
-            }
-        ]
-    }
+    SalesforceClient().send(request)
+
+
+def _save(project_position):
+    position_role = Tag.tags_field_descriptions(project_position.position_role)
+    platform_id__c = project_position.salesforce_job_id()
+    # Skip if the role tag is blank
+    if position_role != '':
+        data = {
+            "GW_Volunteers__Campaign__r":
+                {
+                    "platform_id__c": project_position.position_project.id
+                },
+            "name": position_role,
+            "gw_volunteers__description__c": project_position.position_description
+        }
+        req = requests.Request(
+            method="PATCH",
+            url=f'{client.job_endpoint}/platform_id__c/{platform_id__c}',
+            data=json.dumps(data)
+        )
+        ''' Changed this to a synchronous call to avoid record locks (duplicate position names are possible) '''
+        SalesforceClient().send(req)
```

Comment:
Why is this particular save operation a problem to do asynchronously?

Reply:
Take for example https://www.democracylab.org/projects/1 : Full Stack Developer is listed multiple times. My code comes along and picks up both of these during `merge_chanes()` and we have two updates to the same volunteer_job happening simultaneously--the error is that record locking fails.

Reply:
Is it possible to give it a unique name?  Like "<position name>_<position id>"

Reply:
I believe the Full Stack Developer listings arise from tags so I don't see how to extract position ids

Reply:
If the project_position argument is a [ProjectPosition](https://github.com/DemocracyLab/CivicTechExchange/blob/a39efa91c0ff076513e920abe8a4286b814d61a6/civictechprojects/models.py#L992) object, it should have an id attribute you can use.

Reply:
back to asynchronous save with commit 4a8c88ac0667c4571ed90ba4efcdae9cdd92dad3